File: src/encryption_decryption/csharp/DigitalSignatureExample.cs
using System;
using System.Security.Cryptography;
using System.Text;

public class DigitalSignatureExample {
    public static void Main() {
        using (RSACryptoServiceProvider rsa = new RSACryptoServiceProvider(2048)) {
            string originalData = "Hello, Digital Signatures!";
            byte[] originalDataBytes = Encoding.UTF8.GetBytes(originalData);

            // Sign data
            byte[] signature = rsa.SignData(originalDataBytes, new SHA256CryptoServiceProvider());

            // Verify signature
            bool isVerified = rsa.VerifyData(originalDataBytes, new SHA256CryptoServiceProvider(), signature);

            Console.WriteLine($"Data verified: {isVerified}");
        }
    }
}

File: src/encryption_decryption/python/digital_signature_example.py
from cryptography.hazmat.primitives import hashes
from cryptography.hazmat.primitives.asymmetric import padding, rsa
from cryptography.hazmat.primitives.asymmetric import utils
from cryptography.hazmat.primitives.serialization import load_pem_private_key, load_pem_public_key
from cryptography.hazmat.backends import default_backend

# Generate a new RSA key pair
private_key = rsa.generate_private_key(
    public_exponent=65537,
    key_size=2048,
    backend=default_backend()
)
public_key = private_key.public_key()

# Sign a message
message = b"Hello, digital signatures!"
signature = private_key.sign(
    message,
    padding.PSS(
        mgf=padding.MGF1(hashes.SHA256()),
        salt_length=padding.PSS.MAX_LENGTH
    ),
    hashes.SHA256()
)

# Verify the signature
public_key.verify(
    signature,
    message,
    padding.PSS(
        mgf=padding.MGF1(hashes.SHA256()),
        salt_length=padding.PSS.MAX_LENGTH
    ),
    hashes.SHA256()
)
print("Signature verified!")